plugins/pipelines/metrics.js

```diff
@@ -24,24 +25,33 @@ module.exports = () => ({
         handler: (request, reply) => {
             const factory = request.server.app.pipelineFactory;
             const { id } = request.params;
-            const { aggregateInterval } = request.query;
+            const { aggregateInterval, page, count } = request.query;
             let { startTime, endTime } = request.query;
 
-            if (!startTime || !endTime) {
-                ({ startTime, endTime } = setDefaultTimeRange(startTime, endTime, MAX_DAYS));
-            }
-
             return factory.get(id)
                 .then((pipeline) => {
                     if (!pipeline) {
                         throw boom.notFound('Pipeline does not exist');
                     }
 
-                    if (!validTimeRange(startTime, endTime, MAX_DAYS)) {
-                        throw boom.badRequest(`Time range is longer than ${MAX_DAYS} days`);
-                    }
+                    let config;
 
-                    const config = { startTime, endTime };
+                    // Only when either page or count is unavailable
+                    // check whether startTime and endTime are valid
+                    if (!page || !count) {
+                        if (!startTime || !endTime) {
+                            ({ startTime, endTime } =
+                                setDefaultTimeRange(startTime, endTime, MAX_DAYS));
+                        }
+
+                        if (!validTimeRange(startTime, endTime, MAX_DAYS)) {
+                            throw boom.badRequest(`Time range is longer than ${MAX_DAYS} days`);
+                        }
+
+                        config = { startTime, endTime };
+                    } else {
+                        config = { page, count };
```

Comment:
could set config = { page, count } on line 37 where you declare it then you can remove this `else` statement

Reply:
okay